worker.go

```diff
@@ -0,0 +1,164 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"git.sr.ht/~spc/go-log"
+	"github.com/goccy/go-yaml"
+	"github.com/redhatinsights/rhc-worker-playbook/internal/ansible"
+	"github.com/redhatinsights/rhc-worker-playbook/internal/config"
+	"github.com/redhatinsights/rhc-worker-playbook/internal/exec"
+	"github.com/redhatinsights/yggdrasil/ipc"
+	"github.com/redhatinsights/yggdrasil/worker"
+)
+
+func rx(
+	w *worker.Worker,
+	addr string,
+	id string,
+	responseTo string,
+	metadata map[string]string,
+	data []byte,
+) error {
+	log.Infof("message received: message-id=%v", id)
+
+	returnURL, has := metadata["return_url"]
+	if !has {
+		return fmt.Errorf("invalid metadata: missing return_url")
+	}
+
+	correlationID, has := metadata["crc_dispatcher_correlation_id"]
+	if !has {
+		return fmt.Errorf("invalid metadata: missing crc_dispatcher_correlation_id")
+	}
```

Comment:
When we do not verify playbook, then it is not necessary to check presence of `crc_dispatcher_correlation_id` in metadata. We could set `correlationId` to empty string in this case.